backend/config.go

```diff
@@ -110,6 +116,50 @@ func NewConfig(creds credentials.TransportCredentials, opts ...gorums.ManagerOpt
 	return cfg
 }
 
+func (cfg *Config) replicaConnected(c replicaConnected) {
+	info, peerok := peer.FromContext(c.ctx)
+	md, mdok := metadata.FromIncomingContext(c.ctx)
+	if !peerok || !mdok {
+		return
+	}
+
+	id, err := GetPeerIDFromContext(c.ctx, cfg)
+	if err != nil {
+		cfg.mods.Logger().Warnf("Failed to get id for %v: %v", info.Addr, err)
+		return
+	}
+
+	replica, ok := cfg.replicas[id]
+	if !ok {
+		cfg.mods.Logger().Warnf("Replica with id %d was not found", id)
+		return
+	}
+
+	replica.(*Replica).md = readMetadata(md)
+
+	cfg.mods.Logger().Debugf("Replica %d connected from address %v", id, info.Addr)
+}
+
+const keyPrefix = "hotstuff-"
+
+func convertMetadata(m map[string]string) metadata.MD {
+	md := metadata.New(nil)
+	for k, v := range m {
+		md.Set(keyPrefix+k, v)
+	}
+	return md
+}
+
+func readMetadata(md metadata.MD) map[string]string {
+	m := make(map[string]string)
+	for k, values := range md {
+		if strings.HasPrefix(k, keyPrefix) && len(values) > 0 {
```

Comment:
Go 1.18 has a new `strings.Cut` function that could replace both `HasPrefix` and `TrimPrefix`. Something like this, I think:
```go
if _, key, ok := strings.Cut(k, keyPrefix); ok && len(values) > 0 {
  m[key] = values[0]
```
Maybe the most important place to do this, but if done elsewhere as well, maybe `Cut` can be more efficient.

Reply:
Suggest to use `Cut`, since we probably want to switch go Go 1.18 soon anyway.